### ISSUE TITLE
Change the Crafting Storage level by right-clicking the block itself with correct Storage component!

### ISSUE THREAD

**Issue by Kanzaji** [AppliedEnergistics/Applied-Energistics-2#6934]
### Describe the feature

It's relayed from Enigmatica Discord as you wanted @Technici4n 😄
Video: https://cdn.discordapp.com/attachments/984527413046312980/1077816524477313074/Enigmatica_9_-_Minecraft_1.19_2023-02-21_23-55-46.mp4
Message from Motoko: https://canary.discord.com/channels/195234321836146688/984527413046312980/1077816526356369469
Your Message Technic 😄: https://canary.discord.com/channels/195234321836146688/984527413046312980/1077866294638018580

More detail for others:
Allow to change the Crafting Storage Level / function by right clicking it with the correct item!
The video linked above showcases what I exactly really well!

### Reasons why it should be considered

I'm with Motoko on this one, it feels like it should be default interraction 😄

**Comment by MuteTiefling** [comment]
Glad you liked it :D in case you want to use the same sound effect, it is `minecraft:block.amethyst_block.hit`

Works on all valid locks of the CPU, converting anything to anything else to subvert the need to pick up, uncraft, recraft, and place. Monitors place facing the side that was clicked. 

I don't have a way to convert to a Crafting Unit though. Maybe a sneak right-click, or maybe nothing.

**Comment by MuteTiefling** [comment]
As a bit of an addon to this, it appears the current sneak right-click actions to split these items back out into their crafting components is hard coded?

![image](https://user-images.githubusercontent.com/9543430/222339013-75ad7cfb-4080-4da6-a22c-832828b75cb4.png)

For example, if I override the above recipe with the one below, sneak right-clicking the co-processing unit will still return an engineering processor instead of the wixie charm. 

![image](https://user-images.githubusercontent.com/9543430/222339236-94c4b3e7-d54b-44cb-ad86-4507ef6a2928.png)

**Comment by Technici4n** [comment]
Yes, they are hardcoded indeed. Not sure what the best solution is. They can be disabled in the config.

**Comment by MuteTiefling** [comment]
Hrm... yeah you'd probably need a special recipe handler to guarantee it works. Alright, I'll just disable in the config for now. Didn't realize that was there :D

**Comment by mindforger** [comment]
Isnt that what datapacks are for? I thought that is default now for recipes.. Or do i miss the version information here?

**Comment by MuteTiefling** [comment]
> Isnt that what datapacks are for? I thought that is default now for recipes.. Or do i miss the version information here?

Data packs allow me to change the shapeless craft. They do not allow me to change the Sneak Right Click recovery of crafting components. Those are hard coded.

**Comment by mindforger** [comment]
Oh yeah i was confused sorry .. I somehow thought that it was a crafting recipe but that was another modpack.

**Comment by MuteTiefling** [comment]
Nice work Kanz! <3
